xbmc/threads/Thread.cpp

```diff
@@ -102,6 +102,7 @@ void CThread::Create(bool bAutoDelete, unsigned stacksize)
 
 bool CThread::IsRunning() const
 {
+  CSingleLock lock(m_CriticalSection);
```

Comment:
I think you are talking about something entirely different.. The only important part is here:

>If you looked at the code, you would see that there is no sync wanted. Assume one thread calls IsRunning() and the result is true. Task switch, thread stops

Is it guaranteed that IsRunning is never called from a different thread when m_ThreadId is being written to in stop? If no, it's undefined behaviour.